lib/stm32/f7/Makefile

```diff
@@ -40,9 +40,9 @@ TGT_CFLAGS	= -Os -g \
 
 ARFLAGS		= rcs
 
-OBJS		= rcc.o gpio.o gpio_common_all.o gpio_common_f0234.o
+OBJS		= flash.o rcc.o gpio.o gpio_common_all.o gpio_common_f0234.o
 
-OBJS		+= rcc_common_all.o flash_common_f234.o flash_common_f24.o
```

Comment:
There are a lot in common, but there are also many small differences scattered throughout the registers, and I didn't really have time to look at reference manuals for every series to see what's common. More importantly, I don't think people adding definitions to the common files in the future will actually look at every single reference manual to decide where to add their definitions, and we'll probably end up with extraneous and/or wrong things being defined for some families.

As an example, at the moment we are including flash_common_f24.h, which has a lot of wrong definitions for f7.

In this case I would advocate having a totally separate file because -
1. If it's correct (accurate reflection of the reference manual, which I believe it is now), it's unlikely that we will update it much in the future. This is all pretty low level stuff.
2. It's more readable. People only have to look at one file to see everything that's defined for the family.
3. It's more likely to stay correct, and not be broken by people making changes to flash_common_f12347.h without actually checking the reference manual of every single family.
4. It's easier to verify that it's correct, by just comparing one file to the reference manual for one family.

Downsides:
1. Code duplication means changes to shared parts need to be applied multiple times. But to update a shared file correctly would require looking at every reference manual, which is a lot of work. If people are actually doing that, is it much more work to update each header, too?

In any case, even if we still want to refactor, we can do that later and get this merged first? Since this is at least correct (AFAIK), and the current version isn't.